config/initializers/rack_attack.rb

```diff
@@ -1,21 +1,32 @@
 class Rack::Attack
-  # Throttle login attempts by IP address
-  throttle('logins/ip', limit: proc { CheckConfig.get('login_rate_limit', 10, :integer) }, period: 60.seconds) do |req|
-    if req.path == '/api/users/sign_in' && req.post?
-      req.ip
-    end
+  # Throttle all graphql requests by IP address
+  throttle('api/graphql', limit:  proc { CheckConfig.get('api_rate_limit', 100, :integer) }, period: 60.seconds) do |req|
+    req.ip if req.path == '/api/graphql'
+  end
+
+  # Blocklist IP addresses that are permanently blocked
+  blocklist('block aggressive IPs') do |req|
+    Rails.cache.fetch("block:#{req.ip}") { false }
   end
 
-  # Throttle login attempts by email address
-  throttle('logins/email', limit: proc { CheckConfig.get('login_rate_limit', 10, :integer) }, period: 60.seconds) do |req|
+  # Track excessive login attempts for permanent blocking
+  track('track excessive logins/ip') do |req|
     if req.path == '/api/users/sign_in' && req.post?
-      # Return the email if present, nil otherwise
-      req.params['user']['email'].presence if req.params['user']
+      # Increment the counter for the IP and check if it should be blocked
+      count = Rails.cache.increment("track:#{req.ip}") || 0
+      Rails.cache.write("track:#{req.ip}", count, expires_in: 1.hour)
+
+      # Permanently block if count exceeds the threshold
+      if count >= CheckConfig.get('login_block_limit', 100, :integer)
+        Rails.cache.write("block:#{req.ip}", true)  # No expiration
```

Comment:
@jayjay-w , is this a key defined by `Rack::Attack`? I'm asking because I don't see in the code where the login attempt is explicitly blocked based on this Redis key.

Reply:
Ah, gotcha! Yes, the comment could be more clear, but I understand it now.